test/actions/channels.test.js

```diff
@@ -1832,6 +1832,22 @@ describe('Actions.Channels', () => {
         assert.deepEqual(result, {data: [TestHelper.basicChannel]});
     });
 
+    it('autocompleteChannelsForSearch', async () => {
+        const prefix = TestHelper.basicChannel.name.slice(0, 5);
+
+        nock(Client4.getTeamRoute(TestHelper.basicChannel.team_id)).
+            get('/channels/search_autocomplete').
+            query({name: prefix}).
+            reply(200, [TestHelper.basicChannel]);
+
+        const result = await Actions.autocompleteChannelsForSearch(
```

Comment:
Please change this
```js
const result = await store.dispatch(Actions.autocompleteChannelsForSearch(
    TestHelper.basicChannel.team_id,
    prefix
));
```

We need to stop doing `action()(dispatch, getState)` as that is an anti-pattern

Reply:
I fixed this for all (dispatch, getState) calls in the channels.test.js